### PR TITLE
fix-transferRole-check-on-esdtTransfer

### PR DESCRIPTION
**Pull request by sasurobert**
integrated new flag

### CODE REVIEW

**Review by andrei-marinica — DISMISSED** [review]

**Review by dragos-rebegea — APPROVED** [review]

File: genesis/process/shardGenesisBlockCreator.go [hunk 1]
```diff
@@ -349,17 +350,19 @@ func createProcessorsForShardGenesisBlock(arg ArgsGenesisBlockCreator, enableEpo
...
+		EnableUserNameChange:           false,
+		Marshalizer:                    arg.Core.InternalMarshalizer(),
+		Accounts:                       arg.Accounts,
+		ShardCoordinator:               arg.ShardCoordinator,
+		EpochNotifier:                  epochNotifier,
+		ESDTMultiTransferEnableEpoch:   enableEpochs.ESDTMultiTransferEnableEpoch,
+		ESDTTransferRoleEnableEpoch:    enableEpochs.ESDTTransferRoleEnableEpoch,
+		GlobalMintBurnDisableEpoch:     enableEpochs.GlobalMintBurnDisableEpoch,
+		ESDTTransferMetaEnableEpoch:    enableEpochs.BuiltInFunctionOnMetaEnableEpoch,
+		OptimizeNFTStoreEnableEpoch:    enableEpochs.OptimizeNFTStoreEnableEpoch,
```
Comment (iulianpascalau) [inline genesis/process/shardGenesisBlockCreator.go:364]: this was a forgotten flag. Does the code is still backwards compatible now?

File: go.mod [hunk 2]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/ElrondNetwork/elrond-go-core v1.1.15
 	github.com/ElrondNetwork/elrond-go-crypto v1.0.1
 	github.com/ElrondNetwork/elrond-go-logger v1.0.5
-	github.com/ElrondNetwork/elrond-vm-common v1.2.12
+	github.com/ElrondNetwork/elrond-vm-common v1.2.14-0.20220421185714-edf68ab67a9e
```
Comment (iulianpascalau) [inline go.mod:15]: proper release

**Review by popenta — APPROVED** [review]
@@                            Log scanner                                  @@
##                            fix-transferRole-check-on-transfer           ##
================================================================================
- Known Warnings              450
- New Warnings                342
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Looks good :+1: 
We will wait until the `feat/release-2022-April` gets merged and then re-target this in `rc/2022-may` 
Requested changes only to prevent accidental merges

File: genesis/process/shardGenesisBlockCreator.go (hunk 1 above)
Comment (iulianpascalau) [reply genesis/process/shardGenesisBlockCreator.go:364]: having a second thought, let's keep it this way as this is the correct way. Will deal with the import-db process afterwards

**Review by iulianpascalau — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**